src/json-schema-content-provider.ts

```diff
@@ -3,14 +3,37 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { TextDocumentContentProvider, Uri, ProviderResult, workspace } from 'vscode';
+import { TextDocumentContentProvider, Uri, workspace, window } from 'vscode';
 import { xhr, configure as configureHttpRequests, getErrorStatusDescription, XHRResponse } from 'request-light';
 import { JSONSchemaCache } from './json-schema-cache';
+import { SchemaExtensionAPI } from './schema-extension-api';
 
 export class JSONSchemaDocumentContentProvider implements TextDocumentContentProvider {
-  constructor(private readonly schemaCache: JSONSchemaCache) {}
-  provideTextDocumentContent(uri: Uri): ProviderResult<string> {
-    return getJsonSchemaContent(uri.toString().replace('json-schema://', 'https://'), this.schemaCache);
+  constructor(private readonly schemaCache: JSONSchemaCache, private readonly schemaApi: SchemaExtensionAPI) {}
+  async provideTextDocumentContent(uri: Uri): Promise<string> {
+    if (uri.fragment) {
+      const origUri = uri.fragment;
+      const schemaUri = Uri.parse(origUri);
+      if (origUri.startsWith('https') || origUri.startsWith('http')) {
```

Comment:
Why check startsWith for https and http? They both start with http. This is redundant.